STAT/AliPainter.cxx

```diff
@@ -573,241 +420,577 @@ std::map<TString, Double_t> AliPainter::statValues;
+  if (hisN == nullptr) {
+    ::Error("AliPainter::DrawHistogram", "Input histogram is null");
+    return;
+  }
+  if (pad == nullptr && gPad == nullptr) {
+    ::Error("AliPainter::DrawHistogram", "TPad object doesn't exist.");
+    return;
+  }
+  else if(pad == nullptr && gPad != nullptr) {
+    pad = (TPad *) gPad;
+  }
+  else pad->cd();
+
+  TPad *nextPad = pad;
+  const Int_t kNotDraw = 1<<9;
+
+  TObjArray *hisArr = AliPainter::PrepareHistogram(hisN, expression, keepArray, metaData, verbose);
+  Int_t histCnt = hisArr->GetEntriesFast();
+  TH1D *hisArray[histCnt];
+  for (Int_t i = 0; i < histCnt; i++) {
+    hisArray[i] = (TH1D *) hisArr->At(i);
+    if (hisArray[i]->GetFunction(AliPainter::fitValues["name"]) != nullptr)
+      hisArray[i]->GetFunction(AliPainter::fitValues["name"])->ResetBit(kNotDraw); //delete 0 option for fit
+    if (verbose == 4)
+      ::Info("AliPainter::DrawHistogram", "Histogram %s in processing",hisArray[i]->GetName());
+    if(AliPainter::drawValues["div"] != TString()) {
+      hisArray[i]->Draw();
+      nextPad = AliPainter::GetNextPad(nextPad, nextPad, verbose);
+      if (verbose == 4)
+      nextPad->cd();
+    }
+    else {
+      if (verbose == 4)
+        ::Info("AliPainter::DrawHistogram", "Histogram %s will be draw on the same pad",hisArray[i]->GetName());
+      hisArray[i]->Draw(TString::Format("%sSAME", AliPainter::drawValues["drawOpt"].Data()));
+    }
+  }
+}
+//TODO: think how to combine  AliPainter::ParseString and AliPainter::ParseOptionString @Boris
+//NOTE: we can provide list of brackets and remove ignoreBrackets from arguments add just flag about do you want to ignore brackets or not.
+///
+/// \param inpString
+/// \param sep - separator
+/// \param verbose
+/// \return - array of input strings
+std::vector<TString> AliPainter::ParseString(const char *inpString, const char *sep, Int_t verbose) {
+  std::vector<TString> atts;
+  TString exprsn = TString(inpString);
+  if (exprsn.CountChar('(') != exprsn.CountChar(')')) {
+    ::Error("AliPainter::DrawHistogram", "check brackets in %s", exprsn.Data());
+    return atts;
+  }
+
+  atts.push_back(exprsn(0, exprsn.Index("(", 0)));
+
+  TString verbStr = "";
+  Int_t match = 0, startIndex = 0, finishIndex = 0;
+  Bool_t isChange = kFALSE;
+  for (Int_t i = 0; i < exprsn.Length(); i++) {
+    if (exprsn(i) == sep[0] && match == 0) {
+    match++;
+    startIndex = i;
+    isChange = kTRUE;
+    }
+    else if (exprsn(i) == sep[0] && match > 0) match++;
+    else if (exprsn(i) == sep[1] && match == 1) {
+      match--;
+      finishIndex = i;
+    }
+    else if (exprsn(i) == sep[1] && match > 1) match--;
+
+    if (match == 0 && isChange) {
+      atts.push_back(TString(exprsn(startIndex + 1, finishIndex - startIndex - 1)));
+      isChange = kFALSE;
+    }
+  }
+
+  if (verbose == 4) {
+    TString infoString = "";
+    for (auto const& value: atts)
```

Comment:
This is a C++11 construct as the error message says, and it's not allowed since we - quite unfortunately - need to have this code work with the native SLC6 compiler. Only C++98 is possible here.

Reply:
Ah, ok, thanks!